tests/integration_tests/test_kernel_commandline_match.py

```diff
@@ -75,18 +90,51 @@ def test_lxd_datasource_kernel_override(
     ) in override_kernel_cmdline(ds_str, client)
 
 
+GH_REPO_PATH = "https://raw.githubusercontent.com/canonical/cloud-init/main/"
+
+
 @pytest.mark.skipif(PLATFORM != "lxd_vm", reason="Modifies grub config")
 @pytest.mark.lxd_use_exec
-@pytest.mark.parametrize("ds_str", ("ci.ds=nocloud-net",))
+@pytest.mark.parametrize(
+    "ds_str",
+    (f"ds=nocloud-net;s={GH_REPO_PATH}tests/data/kernel_cmdline_match/",),
+)
 def test_lxd_datasource_kernel_override_nocloud_net(
-    ds_str, client: IntegrationInstance
+    ds_str, session_cloud: IntegrationCloud
 ):
     """NoCloud requirements vary slightly from other datasources with parsing
     nocloud-net due to historical reasons. Do to this variation, this is
     implemented in NoCloud's ds_detect and therefore has a different log
     message.
     """
-
-    assert (
-        "Machine is running on DataSourceNoCloud [seed=None][dsmode=net]."
-    ) in override_kernel_cmdline(ds_str, client)
+    _ds_name, _, seed_url = ds_str.partition(";")
+    _key, _, url_val = seed_url.partition("=")
+    source = get_validated_source(session_cloud)
+    with session_cloud.launch(
+        launch_kwargs={
+            # On Jammy and above, we detect the LXD datasource using a
+            # socket available to the container. This prevents the socket
+            # from being exposed in the container, so LXD will not be detected.
+            # This allows us to wait for detection in 'init' stage with
+            # DataSourceNoCloudNet.
+            "config_dict": {"security.devlxd": False},
+            "wait": False,  # to prevent cloud-init status --wait
+        }
+    ) as client:
+        # We know this will be an LXD instance due to our pytest mark
+        client.instance.execute_via_ssh = False  # pyright: ignore
+        assert wait_for_cloud_init(client, num_retries=60).ok
+        if source.installs_new_version():
+            client.install_new_cloud_init(
+                source, take_snapshot=False, clean=False
+            )
+        logs = override_kernel_cmdline(ds_str, client)
+        assert (
+            "nocloud"
+            == client.execute("cloud-init query platform").stdout.strip()
+        )
+        assert url_val in client.execute("cloud-init query subplatform").stdout
+        assert (
+            "Possible environment: DataSourceNoCloudNet [seed=None]"
```

Comment:
I'm still confused as to why `seed=None`. Shouldn't it be the seed specified on the cli?

Reply:
I understand now. 

Nit, but it might be worth looking for the later log that would include the seed. On my test machine it looked like
`2023-06-26 19:11:46,414 - stages.py[INFO]: Loaded datasource DataSourceNoCloudNet - DataSourceNoCloudNet [seed=cmdline,http://127.0.0.1:8000/][dsmode=net]`

Not priority atm though.

Reply:
This is because self._get_data isn't called until after the log message is emitted.